core/src/main/java/nl/nn/adapterframework/jdbc/migration/DatabaseMigratorBase.java

```diff
@@ -0,0 +1,134 @@
+/*
+Copyright 2017, 2020, 2021 WeAreFrank!
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package nl.nn.adapterframework.jdbc.migration;
+
+import java.io.InputStream;
+import java.io.Writer;
+import java.net.URL;
+import java.sql.SQLException;
+
+import javax.naming.NamingException;
+import javax.sql.DataSource;
+
+import org.apache.logging.log4j.Logger;
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.context.ApplicationContext;
+import org.springframework.jdbc.datasource.TransactionAwareDataSourceProxy;
+
+import lombok.Getter;
+import lombok.Setter;
+import nl.nn.adapterframework.configuration.Configuration;
+import nl.nn.adapterframework.configuration.ConfigurationMessageEvent;
+import nl.nn.adapterframework.configuration.classloaders.ClassLoaderBase;
+import nl.nn.adapterframework.core.IConfigurationAware;
+import nl.nn.adapterframework.jdbc.IDataSourceFactory;
+import nl.nn.adapterframework.jdbc.JdbcException;
+import nl.nn.adapterframework.jndi.JndiDataSourceFactory;
+import nl.nn.adapterframework.util.AppConstants;
+import nl.nn.adapterframework.util.LogUtil;
+
+/**
+ * DatabaseMigration implementation for IAF. 
+ * 
+ * @author	Niels Meijer
+ * @since	7.0-B4
+ *
+ */
+public abstract class DatabaseMigratorBase implements IConfigurationAware, InitializingBean {
+
+	protected Logger log = LogUtil.getLogger(this);
+	private @Setter IDataSourceFactory dataSourceFactory = null;
+	private Configuration configuration;
+	private @Setter String defaultDatasourceName = JndiDataSourceFactory.GLOBAL_DEFAULT_DATASOURCE_NAME;
+	private @Getter String name;
+	private @Getter ClassLoader configurationClassLoader = null;
+	private @Setter String datasourceName;
+
+	@Override
+	public void afterPropertiesSet() {
+		configurationClassLoader = configuration.getClassLoader();
+		if(!(configurationClassLoader instanceof ClassLoaderBase)) { //Though this should technically never happen.. you never know!
+			throw new IllegalStateException("unable to initialize database migrator");
```

Comment:
Ik zou er dan toch bij zetten waarom.

Reply:
Waarom het niet zou moeten kunnen of waarom het moet?